src/NServiceBus.AcceptanceTests/Core/DependencyInjection/When_resolving_before_endpoint_is_started.cs

```diff
@@ -43,6 +44,50 @@ public async Task Should_be_able_to_use_transport_adddress_resolver_after_the_en
         Assert.That(translatedAddress, Is.Not.Null);
     }
 
+    [Test]
+    public async Task Attempt_to_resolve_transport_address_before_endpoint_is_started_throws_meaningful_exception()
+    {
+        var serviceCollection = new ServiceCollection();
+        Exception thrownException = null;
+
+        var result = await Scenario.Define<Context>()
+            .WithEndpoint<ExternallyManagedContainerEndpoint>(b =>
+            b.ToCreateInstance(
+                config => EndpointWithExternallyManagedContainer.Create(config, serviceCollection),
+                async (configured, ct) =>
+                {
+                    var serviceProvider = serviceCollection.BuildServiceProvider();
+
+                    // HINT: Resolve before start
+                    var transportAddressResolver = serviceProvider.GetRequiredService<ITransportAddressResolver>();
+
+                    try
+                    {
+                        // HINT: Call before start
+                        transportAddressResolver.ToTransportAddress(new QueueAddress("SomeAddress"));
```

Comment:
Is there a reason to not use `Assert.Throws<>` here?

Reply:
I wonder if it's worth having some different abstraction to test endpoint configuration. i.e. I give you an endpoint builder, and a context, and then you give me back a fully configured endpoint configuration.

Reply:
It wouldn't bubble but you can set the ex != null as a done statement?

Reply:
Done